GetStream/stream-chat-android
# Task/add ways to include current user in channel avatars

### 🎯 Goal

As per https://getstream.slack.com/archives/C1GRMTJG3/p1669740602535929. 

### 🛠 Implementation details

Added a flag that enables/disables exclusion of the current user in the channel avatar.

### 🎨 UI Changes

For the relevant changes, also check the updated documentation with the customization.

| Before | After |
| --- | --- |
| ![group before](https://user-images.githubusercontent.com/17215808/204732200-879f79f7-4fa4-4e90-8949-20835c25ace8.png) | ![group after](https://user-images.githubusercontent.com/17215808/204732224-c2ae4f66-5dff-4c0a-bdcb-7e068c617ccf.png) |

Bear in mind, 1:1 channels aren't affected.

![1 on 1](https://user-images.githubusercontent.com/17215808/204732275-307b571e-f093-4fba-9546-022916fc902b.png)

### 🧪 Testing

1. Run the UI Sample app - things should work like before.

2. Apply the patch below and do the same - channels should show the current user image too.

<details>

<summary>Apply to override excluding the avatar</summary>

```
Index: stream-chat-android-ui-common/src/main/res/values/attrs_avatar_view.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-common/src/main/res/values/attrs_avatar_view.xml b/stream-chat-android-ui-common/src/main/res/values/attrs_avatar_view.xml
--- a/stream-chat-android-ui-common/src/main/res/values/attrs_avatar_view.xml	(revision c24b393ebcba0028a441001c39d4bcb1a3be6b8b)
+++ b/stream-chat-android-ui-common/src/main/res/values/attrs_avatar_view.xml	(date 1669792324862)
@@ -15,5 +15,5 @@
     limitations under the License.
 -->
 <resources>
-    <bool name="stream_ui_excludeCurrentUserFromChannelAvatars">true</bool>
+    <bool name="stream_ui_excludeCurrentUserFromChannelAvatars">false</bool>
 </resources>
\ No newline at end of file

```
</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
